src/components/CardEditor.tsx

```diff
@@ -1,55 +1,43 @@
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useState, useRef } from "react";
 import "codemirror/lib/codemirror.css";
 import "codemirror/theme/ayu-mirage.css";
 import "codemirror/mode/xml/xml";
 import { UnControlled as CodeMirror } from "react-codemirror2";
 import PROGRAM from "../ast/PROGRAM";
 import Tokenizer from "../lib/tokenizer";
 import { deckCreationLiterals } from "../lib/constants";
-import { cardEditorStrKey } from "../App";
-import { initialCodeEditorStr } from "../";
 import { getHighlights } from "../lib/highlighter";
+import { useDatabase } from "@nozbe/watermelondb/hooks";
+import reconcile from "../lib/reconciler";
+import { debug, debugDB } from "../lib/utils";
+import { astStrKey, cardEditorStrKey } from "../lib/getIintialData";
 
-type Props = { dispatch; createDSLValue: string };
+const literals = ["create deck", "(", ":", ")"];
 
-const getInitialState = () => {
-  const cardStr = localStorage.getItem(cardEditorStrKey);
-  if (cardStr) {
-    return cardStr;
-  } else {
-    const initialVal = `Create Deck Practice Final:
-(1) Foo : Bar
-(2) Bill : Gates
-(3) Steve : Jobs
-(4) Justin : Trudeau 
-(5) Evan : You
-`;
-    localStorage.setItem(cardEditorStrKey, initialVal);
-    return initialVal;
-  }
+function highlight(doc, value, literals) {
+  const highlights = getHighlights(value, literals);
+  highlights.forEach((highlight) => {
+    doc.markText(
+      { line: highlight.lineNumber, ch: highlight.charStart },
+      { line: highlight.lineNumber, ch: highlight.charEnd },
+      { className: "syntax-highlight" }
+    );
+  });
+}
```

Comment:
refactored; highlighting now happens for first load too (it used to only be triggered by subsequent changes)